### ISSUE TITLE
Changing heading type is broken

### ISSUE THREAD

**Issue by 0nko** [wordpress-mobile/WordPress-Aztec-Android#122]
Changing the existing heading type to another in HTML causes unexpected behavior.

### Reproduced
1. Switch to HTML mode and start with `<h1>Heading</h1>`
2. Switch to visual mode; looks fine
3. Switch back to HTML and change it to `<h3>Heading</h3>`
4. Switch to visual mode, notice it says `Heading>` and it's too big (probably H1 size applied)
5. Switch to HTML and observe the following code:
```
<h3></h3>
<h1>Heading</h1>
<h3>&gt;</h3>
```

#### Tested
Emulator running Android 7.1.1

**Comment by theck13** [comment]
This might be related to the cursor position changes (https://github.com/wordpress-mobile/WordPress-Aztec-Android/pull/119).  If you follow the steps below, you'll see what I mean.

### Steps
1. Go to ***HTML*** view.
2. Enter `<h1>Heading</h1>`.
3. Place cursor between `1` and `>` in opening tag. (i.e. `<h1`***`|`***`>Heading</h1>`)
4. Go to ***Visual*** view.
5. Notice unknown tag image.
6. Go to ***HTML*** view.
7. Notice output shown below.

```
<h1>
 &gt;Heading
</h1aztec_cursor>
```

**Comment by 0nko** [comment]
You're right! Klym committed a fix today for another issue with the cursor and this is now working correctly, as well 😎

**Comment by 0nko** [comment]
Fixed by #127.
